exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/TraceExporterTest.java

```diff
@@ -15,25 +15,129 @@
+  @Mock private TraceServiceClient mockedTraceServiceClient;
+  @Mock private TraceServiceStub mockedTraceServiceStub;
+
+  @After
+  public void tearDown() {
+    GlobalOpenTelemetry.resetForTest();
+  }
+
+  @SuppressWarnings("ResultOfMethodCallIgnored")
   @Test
   public void createWithConfiguration() {
-    TraceConfiguration configuration = TraceConfiguration.builder().setProjectId("test").build();
-    try {
+    try (MockedStatic<TraceServiceClient> mockedTraceServiceClient =
+        Mockito.mockStatic(TraceServiceClient.class)) {
+      mockedTraceServiceClient
+          .when(() -> TraceServiceClient.create(Mockito.eq(mockedTraceServiceStub)))
+          .thenReturn(this.mockedTraceServiceClient);
+
+      TraceConfiguration configuration =
+          TraceConfiguration.builder()
+              .setTraceServiceStub(mockedTraceServiceStub)
+              .setProjectId(PROJECT_ID)
+              .build();
       SpanExporter exporter = TraceExporter.createWithConfiguration(configuration);
+      assertNotNull(exporter);
+      generateOpenTelemetryUsingTraceExporter(exporter);
+      simulateExport(exporter);
 
+      mockedTraceServiceClient.verify(
+          () -> TraceServiceClient.create(Mockito.eq(mockedTraceServiceStub)));
+      Mockito.verify(this.mockedTraceServiceClient)
+          .batchWriteSpans((ProjectName) Mockito.any(), Mockito.anyList());
+    }
+  }
+
+  @Test
+  public void createWithDefaultConfiguration() {
+    try (MockedStatic<TraceServiceClient> mockedTraceServiceClient =
+            Mockito.mockStatic(TraceServiceClient.class);
+        MockedStatic<ServiceOptions> mockedServiceOptions =
+            Mockito.mockStatic(ServiceOptions.class);
+        MockedStatic<GoogleCredentials> mockedGoogleCredentials =
+            Mockito.mockStatic(GoogleCredentials.class)) {
+      mockedServiceOptions.when(ServiceOptions::getDefaultProjectId).thenReturn(PROJECT_ID);
+      mockedGoogleCredentials
+          .when(GoogleCredentials::getApplicationDefault)
+          .thenReturn(Mockito.mock(GoogleCredentials.class));
+      mockedTraceServiceClient
+          .when(() -> TraceServiceClient.create(Mockito.any(TraceServiceSettings.class)))
+          .thenReturn(this.mockedTraceServiceClient);
+
+      SpanExporter exporter = TraceExporter.createWithDefaultConfiguration();
       assertNotNull(exporter);
-    } catch (IOException ignored) {
+      generateOpenTelemetryUsingTraceExporter(exporter);
+      simulateExport(exporter);
+
+      mockedTraceServiceClient.verify(
+          Mockito.times(1), () -> TraceServiceClient.create((TraceServiceSettings) Mockito.any()));
+      mockedServiceOptions.verify(Mockito.times(1), ServiceOptions::getDefaultProjectId);
+      Mockito.verify(this.mockedTraceServiceClient)
+          .batchWriteSpans((ProjectName) Mockito.any(), Mockito.anyList());
     }
   }
+
+  @Test
+  public void createNoopSpanExporterIfExceptionThrown() {
```

Comment:
Similar comment as above.  You're verifying flush/shutdown/export.

This doesn't really make sure Noop span exporter is created, you're more testing that a failure in construction doesn't cause later failures in export.

Reply:
Agreed, good point. I will rename this test to reflect the scenario.

Although, just to be clear, a failure in construction here will end up resulting in failure code (for export), but not cause additional exceptions.